src/Microsoft.ML.Tokenizers/Model/SentencePieceBaseModel.cs

```diff
@@ -25,11 +25,11 @@ internal SentencePieceBaseModel(ModelProto modelProto, bool addBos = false, bool
             AddBeginningOfSentence = addBos;
             AddEndOfSentence = addEos;
             BeginningOfSentenceToken = modelProto.TrainerSpec.BosPiece ?? "<s>";
-            BeginningOfSentenceId = modelProto.TrainerSpec.BosId <= 0 ? 1 : modelProto.TrainerSpec.BosId;
+            BeginningOfSentenceId = modelProto.TrainerSpec.BosId;
             EndOfSentenceToken = modelProto.TrainerSpec.EosPiece ?? "</s>";
-            EndOfSentenceId = modelProto.TrainerSpec.EosId <= 0 ? 1 : modelProto.TrainerSpec.EosId;
+            EndOfSentenceId = modelProto.TrainerSpec.EosId;
             UnknownToken = modelProto.TrainerSpec.UnkPiece ?? "<unk>";
-            UnknownId = modelProto.TrainerSpec.UnkId < 0 ? 0 : modelProto.TrainerSpec.UnkId;
+            UnknownId = modelProto.TrainerSpec.UnkId;
```

Comment:
What made you end up removing this? what happens now if the proto model doesn't have these ids?